Add build-9x workflow

2moe opened the pull request:
Let github actions build the rust9x toolchain automatically, then pack+compress it to tar.zst and upload it to github releases.

---

The reason for using **tar.zst**, instead of **7z**, is that 7z will give a warning when packing: 
> **WARNING: The name of the file cannot be resolved by the system.**

2moe committed:
ci(actions): add build-9x workflow

2moe committed:
ci(build-9x): use tar.zstd instead of 7z+lzma

2moe committed:
ci(build-9x): url.repo => github.repository

2moe committed:
ci(build-9x): add ZSTD_LV env

2moe committed:
ci(build-9x): ls -> foreach ls

seritools commented:
Hey, thanks for setting it up!

I'll keep it open for now and have a closer look later, when more changes to rust9x itself are necessary. I'm lucky enough to have a fast PC, so creating a stage 2 build takes just a couple of minutes locally.

I'd like to see these changes:

- Trigger the build on [creating a release](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#release), instead of having the build create the release
- Build stage 2 for the dist artifacts. What I do locally is to use `python x.py install`, see the `[install]` area in `config.rust9x.toml`. The command will build and move everything nicely into the designated dir, so it's easy to compress
- Get 7z back running, as it's probably got much wider support than tar.zstd

2moe committed:
ci(build-9x): tar.zst -> 7z, build(stage1) -> install(stage2), trigger by file -> by tag

2moe committed:
ci(build-9x):  fix typo: symlinks -> symlink

2moe commented:
1. Now it's triggered by "tag". Once you create and push a new tag, it is automatically released. In my opinion, triggered by tag would be better than a manual release to trigger it.

2. I changed it to `python x.py install ${{env.BUILD_ARGS}}`

3. About tar.zstd:
   The Win11(23H2) **Explorer.exe** and **C:/Windows/system32/tar.exe** can now decompress the `tar.zst` format. (Win11 23H2 with bsdtar pre-installed.)
  Nevertheless, I changed it back to 7z. 

4. Consider that actions(Standard GitHub-hosted runners) are a bit slow to build.
If you don't plan to use self-hosted runner, then feel free to close this PR.

2moe committed:
ci(build-9x::get_info::comparation): rm single quotes

2moe committed:
ci(build-9x::get_info): $release[1] -> $release[0]

2moe committed:
ci(buld-9x): compression-algorithm: lzma2 -> lzma

seritools commented:
> Consider that actions(Standard GitHub-hosted runners) are a bit slow to build.

yeah that was my other concern ^^ I don't expect too many updates/constant releases, so doing it manually while testing things locally should be fine at the moment. Should it change (e.g. lots of pull requests that need CI testing) additional process can be added.
